src/utils/zod.ts

```diff
@@ -106,6 +106,18 @@ export type ZodTypeCoercible = z.ZodNumber | z.ZodBoolean | z.ZodBigInt | z.ZodD
 
 export const instanceofZodTypeCoercible = (_type: z.ZodTypeAny): _type is ZodTypeCoercible => {
   const type = unwrapZodType(_type, false);
+
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodUnion)) {
+    return type._def.options.every(
+      (option) => instanceofZodTypeLikeString(option) || instanceofZodTypeCoercible(option),
+    );
+  }
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodIntersection)) {
+    return (
+      instanceofZodTypeCoercible(type._def.left) && instanceofZodTypeCoercible(type._def.right)
+    );
+  }
```

Comment:
I'm less sure of whether this makes sense and might remove it.

It says that a value is coercible if it is made from an intersection of coercible values, however I can't think of practical examples in which this makes sense.